tools/@aws-cdk/spec2cdk/test/services.test.ts

```diff
@@ -25,3 +25,147 @@ test('can codegen service with arbitrary suffix', () => {
   expect(rendered).toContain('function convertCfnApplicationV2PropsToCloudFormation');
   expect(rendered).toContain('function CfnApplicationV2ApplicationCodeConfigurationPropertyValidator');
 });
+
+test('resource interface when primaryIdentifier is an attribute', () => {
+  const service = db.lookup('service', 'name', 'equals', 'aws-voiceid').only();
+
+  const ast = AstBuilder.forService(service, { db });
+
+  const rendered = renderer.render(ast.module);
+
+  expect(rendered).toContain([
+    '/**',
+    ' * Shared attributes for both `CfnDomain` and `Domain`.',
+    ' *',
+    ' * @struct',
+    ' * @stability external',
+    ' */',
+    'export interface ICfnDomain {',
+    '  /**',
+    '   * The identifier of the domain.',
+    '   *',
+    '   * @cloudformationAttribute DomainId',
+    '   */',
+    '  readonly attrDomainId: string;',
+    '}',
+  ].join('\n'));
+});
+
+test('resource interface when primaryIdentifier is a property', () => {
+  const service = db.lookup('service', 'name', 'equals', 'aws-kinesisanalyticsv2').only();
+
+  const ast = AstBuilder.forService(service, { db });
+
+  const rendered = renderer.render(ast.module);
+
+  expect(rendered).toContain([
+    '/**',
+    ' * Shared attributes for both `CfnApplication` and `Application`.',
+    ' *',
+    ' * @struct',
+    ' * @stability external',
+    ' */',
+    'export interface ICfnApplication {',
+    '  /**',
+    '   * The name of the application.',
+    '   *',
+    '   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kinesisanalyticsv2-application.html#cfn-kinesisanalyticsv2-application-applicationname',
+    '   */',
+    '  readonly attrApplicationName?: string;', // optional?
+    '}',
+  ].join('\n'));
+});
+
+test('resource interface with multiple primaryIdentifiers', () => {
+  const service = db.lookup('service', 'name', 'equals', 'aws-lakeformation').only();
+
+  const ast = AstBuilder.forService(service, { db });
+
+  const rendered = renderer.render(ast.module);
+
+  expect(rendered).toContain([
+    '/**',
+    ' * Shared attributes for both `CfnDataCellsFilter` and `DataCellsFilter`.',
+    ' *',
+    ' * @struct',
```

Comment:
It's definitely NOT a struct

Reply:
Can i get clarification on what is/isn't a struct? I was confused as to why our property interfaces are typed as `StructType` and not `InterfaceType`, so I copied that behavior over. Right now, all our property interfaces have `@struct` too.

Reply:
Figured the nuances out w @mrgrain